core/drivers/imx/dcp/dcp_utils.c

```diff
@@ -0,0 +1,180 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2020 NXP
+ */
+#include <dcp_utils.h>
+#include <drivers/imx/dcp.h>
+#include <kernel/cache_helpers.h>
+#include <malloc.h>
+#include <mm/core_memprot.h>
+#include <trace.h>
+
+struct mem_hdr {
+	void *addr;
+	size_t size;
+} __packed;
```

Comment:
It looks wrong to have this packed. Why do you need to force the compiler to generate code for unaligned accesses to these fields?

Reply:
It because of ```#define OF_MEM_HDR(p)``` that might mess with the alignment:
```error: cast increases required alignment of target type [-Werror=cast-align]```
```__packed``` attribute discard the error. Something identical has been done the CAAM driver.

Should we wait for https://github.com/OP-TEE/optee_os/pull/4293 ?

Reply:
> Should we wait for #4293 ?

Yes that should make it easier here. That PR should hopefully not take too long to merge. If you have the opportunity please help reviewing that PR. Especially the tricky part in `bget()`.